django_ai_assistant/helpers/assistants.py

```diff
@@ -422,7 +450,36 @@ class AgentState(TypedDict):
             output: str
 
         def setup(state: AgentState):
-            return {"messages": [SystemMessage(content=self.get_instructions())]}
+            messages = [SystemMessage(content=self.get_instructions())]
+
+            if self.structured_output:
+                schema = None
+
+                # If Pydantic
+                if inspect.isclass(self.structured_output) and issubclass(
+                    self.structured_output, BaseModel
+                ):
+                    schema = json.dumps(self.structured_output.model_json_schema())
+
+                schema_information = ""
+                if schema:
+                    schema_information = f"JSON will have the following schema:\n\n{schema}\n\n"
+
+                # The assistant won't have access to the schema of the structured output before
+                # the last step of the chat. This message gives visibility about what fields the
+                # response should have so it can gather the necessary information by using tools.
+                messages.append(
+                    SystemMessage(
+                        content=(
+                            "In the last step of this chat you will be asked to respond in JSON. "
+                            + schema_information
+                            + "Gather information using tools. "
```

Comment:
What if there are no tools? Perhaps check?

Reply:
Do you think "Gather information, use appropriate tools if there are any available. " would make this better? The automated tests I wrote don't have tools and it wasn't a problem.

Reply:
I think we could check if there are any tools or not, then only include that string if there are tools.